module/VuFind/src/VuFind/Controller/CombinedController.php

```diff
@@ -350,11 +350,30 @@ protected function adjustQueryForSettings($settings, $searchType = null)
         } elseif (!$includeRecommendSetting) {
             $noRecommend[] = 'top';
         }
+
+        // Display or hide side based on include_recommendations_side setting.
         if (is_array($settings['include_recommendations_side'] ?? false)) {
             $recommendOverride['side'] = $settings['include_recommendations_side'];
         } else {
             $noRecommend[] = 'side';
         }
+
+        // Display or hide no results recommendations, based on
+        // include_recommendations_noresults setting (to display them in the bento box) or
+        // include_recommendations_noresults_side setting (to display them in the sidebar).
+        $includeRecommendNoResultsSetting = $settings['include_recommendations_noresults'] ?? false;
+        if (is_array($includeRecommendNoResultsSetting)) {
+            $recommendOverride['noresults'] = $settings['include_recommendations_noresults'];
+        } elseif (!$includeRecommendNoResultsSetting) {
+            $noRecommend[] = 'noresults';
+        }
+
+        if (is_array($settings['include_recommendations_noresults_side'] ?? false)) {
+            $recommendOverride['noresults_side'] = $settings['include_recommendations_noresults_side'];
+        } else {
+            $noRecommend[] = 'noresults_side';
```

Comment:
I don't think you strictly need to add `noresults_side` here, since `noresults_side` is not a standard location that will have configuration set under circumstances other than combined search... but it may be worth leaving this anyway just for consistency, and in case we add a noresults sidebar to non-combined search in the future.

Reply:
> and in case we add a noresults sidebar to non-combined search in the future

Yeah I think it's useful to leave for that reason.  There are a number of recommendation modules that work fine whether or not there are any search results -- just using the search query.  I.e. the various AbstractSearchObjects, or LibGuides & AZ.